src/components/DataProvider/index.js

```diff
@@ -105,6 +115,14 @@ export default class DataProvider extends Component {
         await this.fetchData(id, 'UPDATE_SUBDOMAIN');
       }
     });
+
+    // Check if basedomain changed in props -- if so reset state.
+    if (!isEqual(this.props.baseDomain, prevProps.baseDomain)) {
```

Comment:
This is touched on in the discussion -- this is a fallback for the props (will update based on feedback/discussion).